### ISSUE TITLE
FAN Manual Control

### ISSUE THREAD

**Issue by zigastegu** [luftek/LimeSDR-USB#19]
How to check if board drives FAN output correctly?
1. Wait for the temperature to raise above 55°C for FAN to turn on, and below 45°C again to turn OFF
2. Manual option? 

Fan is driven with FAN_CTRL signal from FPGA. FPGA internally has NIOS2 cpu instance that sets temperature module LM75 to hysteresis 45 and 55°C with output signal which is rerouted through FPGA input through internal MUX to FAN_CTRL (E4) output

![image](https://user-images.githubusercontent.com/1379063/28745447-2f61e2ce-7479-11e7-8078-0d4953a7bd5b.png)

**Hardcoded test:**
- periphcfg.vhd reserves memory: 
```
mem(12)	<= "000000000000000x"; --  0 free, PERIPH_OUTPUT_OVRD_0
mem(13)	<= "000000000000000x"; --  0 free, PERIPH_OUTPUT_VAL_0
```
default values are both 0 and further uses LSB bits for driving FAN_CTRL (see photo above)
changing LSB of mem(12) changes source and mem(13) is manual signal 

I have compiled 2 versions based on https://github.com/myriadrf/LimeSDR-USB_GW/commit/0f021ca1b6faf9a09b5736b25f60ee6e562a5822:
and used LimeUtil.exe to flash them on FPGA
[test_FPGA_Fan.zip]()

**WIP:**
I think changing this memory addresses is possible over SPI as LimeSuite GUI implements features for 16bit spi board writing also "cpu" in FPGA handles writing for GPIO
Q:
Main question what is the offset address? Do need to change any other register before writing this address? FPGA SPI Slave Select?

Hopefully digging through implementation of GUI FPGA GPIO will give us answers

**Comment by zigastegu** [comment]
I was so close :)
offsets are:
0xCC - 0 = auto, 1= manual
0xCD - 0 = off, 1 = on

![image](https://user-images.githubusercontent.com/1379063/28746412-2e710008-748c-11e7-97be-2d684d5ceb05.png)

![image](https://user-images.githubusercontent.com/1379063/28746337-9550c0d0-748a-11e7-8384-5962e5fd6716.png)

**Comment by Hyseas** [comment]
I think I am missing something. I went to Board SPI, Write, entered the hex values as shown above, issued a CMD Write command, received Status: Write success.  However, Read, entered the hex values as shown above, issued a CMD Read command and the Read success never varied from Hex: 0x0000.  I could not get values to change or the fan to turn on.  Can you please elaborate on the process to test the fan manually?

**Comment by luftek** [comment]
0xCC - 0 = auto, 1= manual
0xCD - 0 = off, 1 = on

**Comment by Hyseas** [comment]
Thank you for your response. I think I tried that but may still be doing it wrong. Let me provide some additional information.

I have the LimeSDR-USB v1.4 with updated firmware. I'm using Lime Suite GUI v17.02.2-PothosSDR-2017.05.16-vc1. I have gone to Options>ConnectionSettings and selected USB 3.0 (LimeSDR-USB) [USB] 9061C02CA0B20. The log shows the device connected.

I then select Options>SPI>Board SPI (section)>Write>Address (Hex): and enter "0xCC". Then go to Write>Value (Hex): and enter "01". I then enter "CDM Write", which returns "Status: Write success".

I then select Board SPI (section)>Read>Address (Hex): and enter "0xCC" and enter "CDM Read". It returns >Read Value (Hex): 0x0000 and Status: Read Success. (see below)
![image](https://user-images.githubusercontent.com/30727127/28985238-95724bac-792f-11e7-9540-8ca524985548.png)

I also tried the format shown (FFFF) and used Value (Hex): 0001. Still the same result. I really would like to test the fan, so any guidance or direction is greatly appreciated. Thank you.
